include/sdf_tools/sdf.hpp

```diff
@@ -40,7 +40,7 @@ namespace sdf_tools
 
         SignedDistanceField(std::string frame, double resolution, double x_size, double y_size, double z_size, float OOB_value);
 
-        SignedDistanceField(Eigen::Isometry3d origin_transform, std::string frame, double resolution, double x_size, double y_size, double z_size, float OOB_value);
+        SignedDistanceField(const Eigen::Isometry3d& origin_transform, std::string frame, double resolution, double x_size, double y_size, double z_size, float OOB_value);
```

Comment:
Why do we want to pass the origin_transform by reference?
Two potential problems
```
Eigen::Isometry3d& my_origin = something;
SignedDistanceField my_sdf(my_origin, ...);
my_origin = something_else // This changes the behavior of the my_sdf
```
```
SignedDistanceField my_sdf(Eigen::Isometry3d(stuff), ...);
// the origin_transform memory is now no longer allocated
```


I see it is done in VoxelGrid as well.

Reply:
All fixed size vectorizable Eigen types must be passed by reference. https://eigen.tuxfamily.org/dox/group__TopicFixedSizeVectorizable.html
A copy is then made inside of the VoxelGrid constructor. So this is not threadsafe, but that's unaviodable.

Reply:
The local copy mitigates the (non-threaded) concern